src/components/ResourceList/components/FilterControl/components/DateSelector/DateSelector.tsx

```diff
@@ -49,200 +36,296 @@ export enum DateFilterOption {
   OnOrAfter = 'on_or_after',
 }
 
-class DateSelector extends React.PureComponent<CombinedProps, State> {
-  state: State = {
-    datePickerMonth: this.now.getMonth(),
-    datePickerYear: this.now.getFullYear(),
-    initialConsumerFilterKey: this.props.filterKey,
-  };
+// eslint-disable-next-line react/display-name
+export const DateSelector = memo(function DateSelector({
+  filterValue,
+  filterKey,
+  filterMinKey,
+  filterMaxKey,
+  dateOptionType,
+  onFilterValueChange,
+  onFilterKeyChange,
+}: DateSelectorProps) {
+  const now = new Date();
+
+  const {translate} = useI18n();
+  const initialConsumerFilterKey = useLazyRef(() => filterKey);
```

Comment:
What's the value in using useLazyRef here? It looks like filterKey is a plain string. using useLazyRef instead of useRef is useful if the default value is something that would be expensive to compute on every render, but that doesn't seem to be the case here.

Reply:
Your right, it was a blip, no need to lazyRef here 😄